osu.Game/Beatmaps/ControlPoints/DifficultyControlPoint.cs

```diff
@@ -51,11 +44,7 @@ public double SliderVelocity
         /// Whether or not slider ticks should be generated at this control point.
         /// This exists for backwards compatibility with maps that abuse NaN slider velocity behavior on osu!stable (e.g. /b/2628991).
         /// </summary>
-        public bool GenerateTicks
-        {
-            get => GenerateTicksBindable.Value;
-            set => GenerateTicksBindable.Value = value;
-        }
+        public bool GenerateTicks { get; set; } = true;
```

Comment:
I haven't 100% confirmed if it's possible, but if you can move this into `LegacyDifficultyControlPoint`, this change is going to be a *lot* easier to swallow.

Reply:
Looks like I'll have to do this for each mode:
```
#pragma warning disable 618
bool generateTicks = (DifficultyControlPoint as LegacyBeatmapDecoder.LegacyDifficultyControlPoint)?.GenerateTicks ?? true;
#pragma warning restore 618
```

Which looks really really messy. I could...
1. Go and copy and paste this everywhere (gross)
2. Have some property in `HitObject` for it (overkill)
3. Have some interface like `IHasSliderTicks` or something (maybe still overkill?)
4. Ditch porting this behavior to other modes and keep it a standard-only thing (inconsistent)

At this point I'm leaning towards the 4th option.

Reply:
If it doesn't affect any beatmaps in the other rulesets, then option 4 is fine.

Reply:
If I remember correctly, having NaN slider behaviour in stable on rulesets like CTB work so long as it's not a converted beatmap. Only Standard and CTB rulesets would be applied for this change.
In the case of CTB, only the large droplets would be missing whilst the small droplets are still visible but are greatly reduced.

Reply:
Converted CTB maps also seem to remove the large ticks from my testing, but small ticks seem to be unaffected. Are there any CTB maps that actually use this behavior?

Reply:
I think it is worth mentioning, NaN in stable prevents subsequent slider judgements. For example, a NaN slider with a reverse arrow, the slider end will not generate properly, forcing a 100. A NaN slider with 2 reverse arrows will not judge the second arrow or slider end. I think this should be fixed in stable as it is a source of bugs and game crashes there: https://osu.ppy.sh/beatmapsets/948643/discussion/-/generalAll

Also worth mentioning, NaN red line also used to do something in stable, it was changed recently to read as 1000, but it used to keep beat duration from the previous red line and remove ticks. Since this didn't enable new behavior over NaN greenlines, I don't think any maps rely on it and stable was changed anyway.

Reply:
Fixing anything in stable is not in scope of this discussion (and unlikely to happen anyway).